python/delta/integration_tests/unity-catalog-commit-coordinator-integration-tests.py

```diff
@@ -54,54 +63,195 @@
             "org.delta.catalog.UCCoordinatedCommitClient") \
     .getOrCreate()
 
-expected_error_tag = "UNITY_CATALOG_EXTERNAL_COORDINATED_COMMITS_REQUEST_DENIED"
 
+EXTERNAL_CLIENT_DELETE_ERROR = \
+    f"Request to delete non-external table '{CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE}' " \
+    "from outside of Databricks Unity Catalog enabled compute environment is not supported."
+S3_FORBIDDEN_ACCESS_ERROR = ("Forbidden (Service: Amazon S3; Status Code: 403; "
+                             "Error Code: 403 Forbidden;")
+WRITE_FAILED_ERROR = "[TASK_WRITE_FAILED] Task failed while writing rows to s3"
+
+
+def read(table_name) -> DataFrame:
+    return spark.read.table(f"{CATALOG_NAME}.{SCHEMA}.{table_name}")
+
+
+def read_with_timestamp(timestamp, table_name) -> DataFrame:
+    return spark.read.option("timestampAsOf", timestamp).table(
+        f"{CATALOG_NAME}.{SCHEMA}.{table_name}")
+
+
+def read_with_cdf_timestamp(timestamp, table_name) -> DataFrame:
+    return spark.read.option('readChangeFeed', 'true').option(
+        "startingTimestamp", timestamp).table(f"{CATALOG_NAME}.{SCHEMA}.{table_name}")
+
+
+def create_df_with_rows(list_of_rows) -> DataFrame:
+    return spark.createDataFrame(list_of_rows,
+                                 schema=StructType([StructField("id", IntegerType(), True)]))
+
+
+class UnityCatalogCommitCoordinatorTestSuite(unittest.TestCase):
+    setup_df = spark.createDataFrame([(1, ), (2, ), (3, )],
+                                     schema=StructType([StructField("id", IntegerType(), True)]))
+
+    def setUp(self) -> None:
+        self.setup_df.write.mode("overwrite").insertInto(
+            f"{CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE}")
+
+    # DML Operations #
+    def test_update(self) -> None:
+        spark.sql(f"UPDATE {CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE} SET id=4 WHERE id=1")
+        updated_tbl = read(MANAGED_CC_TABLE).toDF("id")
+        assertDataFrameEqual(updated_tbl, create_df_with_rows([(4, ), (2, ), (3, )]))
+
+    def test_delete(self) -> None:
+        spark.sql(f"DELETE FROM {CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE} where id=1")
+        updated_tbl = read(MANAGED_CC_TABLE).toDF("id")
+        assertDataFrameEqual(updated_tbl, create_df_with_rows([(2, ), (3, )]))
+
+    def test_merge(self) -> None:
+        spark.sql(f"MERGE INTO {CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE} AS target "
+                  f"USING (VALUES 2, 3, 4, 5 AS src(id)) AS src "
+                  f"ON src.id = target.id WHEN NOT MATCHED THEN INSERT *;")
+        updated_tbl = read(MANAGED_CC_TABLE).toDF("id")
+        assertDataFrameEqual(updated_tbl, create_df_with_rows([(1, ), (2, ), (3, ), (4, ), (5, )]))
+
+    # Utility Functions #
+    def test_optimize(self):
+        spark.sql(f"OPTIMIZE {CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE}")
+        updated_tbl = read(MANAGED_CC_TABLE).toDF("id")
+        assertDataFrameEqual(updated_tbl, create_df_with_rows([(1, ), (2, ), (3, )]))
```

Comment:
Can you also call DESCRIBE HISTORY here to check that the last operation is OPTIMIZE?